File: src/main/java/com/tram/springbootangularboard/security/token/PreAuthorizationToken.java
package com.tram.springbootangularboard.security.token;

import com.tram.springbootangularboard.dto.FormLoginDto;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;

public class PreAuthorizationToken extends UsernamePasswordAuthenticationToken {
    private PreAuthorizationToken(String userId, String password) {
        super(userId, password);
    }
    public PreAuthorizationToken(FormLoginDto formLoginDto) {
        this(formLoginDto.getUserId(), formLoginDto.getPassword());
    }
    public String getEmail() {
        return (String)super.getPrincipal();
    }

    public String getPassword() {
        return (String)super.getCredentials();
    }
}

File: src/main/java/com/tram/springbootangularboard/security/tokens/PreAuthorizationToken.kt
package com.tram.springbootangularboard.security.tokens

import org.springframework.security.authentication.UsernamePasswordAuthenticationToken

class PreAuthorizationToken : UsernamePasswordAuthenticationToken{
    constructor(principal: kotlin.Any, credentials: kotlin.Any): super(principal, credentials)
    constructor(userId: String, password: String): super(userId, password)
    fun getUserId(): String {
        return super.getPrincipal() as String
    }
    fun getUserPassword(): String {
        return super.getCredentials() as String
    }
}